africanus/experimental/rime/fused/examples/custom_brightness_term.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import numpy as np
+from numba import literal_unroll
+from numba import types
+from numba.core.errors import TypingError
+from numba.cpython.unsafe.tuple import tuple_setitem
+
+
+from africanus.experimental.rime.fused.intrinsics import tuple_fill
+from africanus.experimental.rime.fused.specification import RimeSpecification
+from africanus.experimental.rime.fused.terms.core import Term
+from africanus.experimental.rime.fused.core import rime
+from daskms import xds_from_storage_ms as xds_from_ms
+from daskms import xds_from_storage_table as xds_from_table
+
+
+class ModelFlux(Term):
+    """
+    A custom model flux provider to substitute the standard Brightness term.
+    Here out model flux is provided as an array of shape (nsource, nchan, nstokes)
+    """
+
+    def __init__(self, configuration, stokes):
+        super().__init__(configuration)
+        self.stokes = stokes
+
+    def dask_schema(
+        self,
+        model_flux,
+    ):
+        return {
+            "model_flux": ("source", "chan", "stokes"),
+        }
+
+    def init_fields(self, typingctx, init_state, model_flux):  # (ndir, nchan, nstokes)
+        if not isinstance(model_flux, types.Array) or model_flux.ndim != 3:
+            raise TypingError(
+                f"model_flux {model_flux} should be a (ndir, nchan, nstokes) array"
+            )
+
+        return [], lambda init_state, model_flux: None
```

Comment:
This simplified a bit (I think in the original version we were defining a function (rather than just a dummy lambda).

Actually if `model_flux` is a requested input, it gets placed on the `state` object in the sampler anyway (I'd forgotten this), so there was no need to do this ourselves